### DISCUSSION TITLE
Order of arguments in read_resource

### DISCUSSION THREAD

**Discussion by fjuniorr** [inbo/datapackage#52] [Ideas]
I've noted that the order of the arguments have changed in `read_resource` from `(package, resource_name)` to `(resource_name, package)` in https://github.com/frictionlessdata/frictionless-r/commit/ba943b13e123d0b05575fb036054f5ac53e4c17a.

One of the advantages of the previous ordering is that we could write code with both base (`|>`) and magrittr (`%>%`) pipes in the following style:

```R
package |> read_resource("resource-name")
```

With the new order, with magrittr pipes we can still use `.` as placeholder

```R
# still not as pleasant
package %>% read_resource("resource-name", .) 
```

but with the base pipe operator I need

```R
# not fun
package |> (function(x) read_resource("resource-name", x))() 
```

I couldn't find the reasoning behind the change, but wanted to chime in in the discussion for the design of [`frictionless-r`](https://github.com/frictionlessdata/frictionless-r).

**Comment by peterdesmet** [comment]
Hi @fjuniorr, thanks for the suggestion!

The switch was initially [suggested](https://github.com/frictionlessdata/frictionless-r/issues/11#issuecomment-860780848) by my colleague @damianooldoni when thinking about write functions. He found _read resource name (1) from package (2)_ a more natural way of expressing the arguments. I [concurred](https://github.com/frictionlessdata/frictionless-r/issues/11#issuecomment-860789412) that having e.g. the dataframe first would work well with piping.

However, you make a good case for piping with `package`. In fact, many of the functions return `package`:

```
create_package()                        -> package
create_schema(df)                       -> schema (more a helper function)
get_resource(resource_name, package)    -> resource (more a helper function)
get_schema(resource_name, package)      -> schema (more a helper function)
read_package(file)                      -> package
read_resource(resource_name, package)   -> df
remove_resource(resource_name, package) -> package
write_package(package, directory)       -> package
```

Some, in addition to:

```R
read_package() %>% read_resource("resource-name")
```

The following scenario will also supported if we make `package` the first argument:

```R
create_package() %>%
  add_resource(df1) %>% # function to be developed
  add_resource(df2) %>%
  write_package("directory")
```

So, I'll revert the switch and make `package` the first argument where we can. Supporting piping also aligns well with:

> Frictionless is an R package to read and write Frictionless Data Packages in a Tidyverse way.
